jobs/post_persistent_notifications/scheduler.go

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package post_persistent_notifications
+
+import (
+	"time"
+
+	"github.com/mattermost/mattermost-server/v6/jobs"
+	"github.com/mattermost/mattermost-server/v6/model"
+)
+
+type Scheduler struct {
+	*jobs.PeriodicScheduler
+}
+
+func (scheduler *Scheduler) NextScheduleTime(cfg *model.Config, _ time.Time, _ bool, _ *model.Job) *time.Time {
+	nextTime := time.Now().Add((time.Duration(*cfg.ServiceSettings.PersistentNotificationInterval) * time.Minute) / 2)
```

Comment:
`/2` is actually to minimize the chance of missing new PNs when job has just run and a new PN comes, then it will have to wait almost twice of the original `PersistentNotificationInterval` before sending the first PN. It basically narrows the window of new PNs coming in.

Haven't discussed the job _sleeping_ idea. Not sure how that works, but for our case it may not even be suitable. As it's not just that the job can just wake-up whenever a new PN post comes but it needs to keep running persistently to keep sending the PN for older posts as well. 

It would be great, if you know any sleeping-job example in the codebase. I'll see if our use-case can be fulfilled with it or not.

Reply:
It's probably not helpful in this case, but we tried something similar over at https://github.com/mattermost/mattermost-plugin-api/blob/24f6408ed7a6a6830e8c3060ba9c865dbbe40335/cluster/job.go#L197-L213.